python/ray/data/_internal/arrow_block.py

```diff
@@ -175,6 +175,13 @@ def _build_tensor_row(row: ArrowRow) -> np.ndarray:
         # Getting an item in a tensor column automatically does a NumPy conversion.
         return row[VALUE_COL_NAME][0]
 
+    def get_keys(self, key: KeyFn) -> Optional[np.ndarray]:
+        # The key must be a str for Arrow table.
+        assert isinstance(key, str)
+        keys = self._table[key].to_pandas().to_numpy()
```

Comment:
I think a direct `.to_numpy()` call should work?
```suggestion
        keys = self._table[key].to_numpy()
```

Reply:
oh nice, didn't realize it has this interface. Now we can merge pandas/arrow into one shared get_keys().